pkg/resources/view_grant.go

```diff
@@ -107,14 +110,23 @@ func CreateViewGrant(data *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	// ID format is <db_name>|<schema_name>|<view_name>|<privilege>
-	// view_name is empty when on_future = true
-	if futureViews {
-		data.SetId(fmt.Sprintf("%v|%v||%v", dbName, schemaName, priv))
-	} else {
-		data.SetId(fmt.Sprintf("%v|%v|%v|%v", dbName, schemaName, viewName, priv))
+	data_identifiers := [][]string{{dbName, schemaName, viewName, ""}}
+
+	if !(futureViews) { // merge with above if/else statement if possible
+		data_identifiers[0][3] = priv
 	}
 
+	var buf bytes.Buffer
+	csv_writer := csv.NewWriter(&buf)
+	csv_writer.Comma = '|'
+	csv_writer.WriteAll(data_identifiers)
+
+	if err := csv_writer.Error(); err != nil {
+		log.Fatalln("error writing csv:", err)
```

Comment:
this will abruptly exit your code and call something like `os.Exit(1)`. https://github.com/golang/go/blob/go1.13.1/src/log/log.go#L329:6

Reply:
This function allows you to return an error `func CreateViewGrant(data *schema.ResourceData, meta interface{}) error {`
so you can just `return err` and have it be someone else's problem to deal with